src/test/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReaderTest.java

```diff
@@ -313,6 +317,56 @@ private void assertRecordReaderOutputRecords(String[] csvStrings, Schema schema,
     }
   }
 
+  /**
+   * The FailsafeException is thrown with the expected error message.
+   * The retry policy is correctly configured with the specified configurations.
+   */
+  @Test(expected = SalesforceQueryExecutionException.class)
+  public void testRetryMechanism() throws Exception {
```

Comment:
When testing retry logic with `JUnit`, testing both the failure scenario that triggers a retry and the success scenario after a certain number of retries is a good practice. 

For example, see: https://github.com/cdapio/cdap/blob/1be1fa025be3d2af490665bcc6bac0c19649d24b/cdap-data-fabric/src/test/java/io/cdap/cdap/spi/data/sql/SqlStructuredTableRetryTest.java#L45-L53

Reply:
Added